Cannot run client in dev due to space in my Windows file path

Attempting to run Neoforge with a setup very close to the MDK, I get this error:
```
Connected to the target VM, address: '127.0.0.1:63513', transport: 'socket'
Error: Could not find or load main class Laptop\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm-commons\9.5\19ab5b5800a3910d30d3a3e64fdb00fd0cb42de0\asm-commons-9.5.jar;C:\Users\MSI
Caused by: java.lang.ClassNotFoundException: Laptop\/gradle\caches\modules-2\files-2/1\org/ow2/asm\asm-commons\9/5\19ab5b5800a3910d30d3a3e64fdb00fd0cb42de0\asm-commons-9/5/jar;C:\Users\MSI
Disconnected from the target VM, address: '127.0.0.1:63513', transport: 'socket'
```

My file path for my project is:
`C:\Users\MSI Laptop\Documents\ModWorkspace\CommandStructures-Neoforge`

The space seems to be what is breaking the project. Not sure if I can remove that space without breaking a bunch of my other non-mc stuff since I had that path since like forever.

I'm having a similar problem trying to run neoforgedev:

* cloned fork of neoforge to `C:\Users\Joseph\Saved Games\minecraft mods\neoforge`
* ran `gradlew :neoforge:setup`
* import project/repo folder into intellij
* Run neoforge client

Results in this error:

```
"C:\Program Files\Eclipse Adoptium\jdk-17.0.8.7-hotspot\bin\java.exe" -p C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\cpw.mods\securejarhandler\2.1.10\51e6a22c6c716beb11e244bf5b8be480f51dd6b5\securejarhandler-2.1.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm\9.5\dc6ea1875f4d64fbc85e1691c95b96a3d8569c90\asm-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm-commons\9.5\19ab5b5800a3910d30d3a3e64fdb00fd0cb42de0\asm-commons-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm-tree\9.5\fd33c8b6373abaa675be407082fdfda35021254a\asm-tree-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm-util\9.5\64b5a1fc8c1b15ed2efd6a063e976bc8d3dc5ffe\asm-util-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm-analysis\9.5\490bacc77de7cbc0be1a30bb3471072d705be4a4\asm-analysis-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\cpw.mods\bootstraplauncher\1.1.2\c546e00443d8432cda6baa1c860346980742628\bootstraplauncher-1.1.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged\JarJarFileSystems\0.4.0\ef7e5716525bbe50c784a362f9393457a33e6daf\JarJarFileSystems-0.4.0.jar --add-modules ALL-MODULE-PATH --add-opens java.base/java.util.jar=cpw.mods.securejarhandler --add-opens java.base/java.lang.invoke=cpw.mods.securejarhandler --add-exports java.base/sun.security.util=cpw.mods.securejarhandler --add-exports jdk.naming.dns/com.sun.jndi.dns=java.naming -Djava.net.preferIPv6Addresses=system -DignoreList=securejarhandler,asm,asm-commons,asm-tree,asm-util,asm-analysis,bootstraplauncher,JarJarFileSystems,events-1.0.10.jar,core-1.0.10.jar,language-java,language-lowcode,language-minecraft,client-extra,neoforge- -DmergeModules=jna-5.10.0.jar,jna-platform-5.10.0.jar -Dfml.pluginLayerLibraries=core-1.0.10.jar,language-java-1.0.10.jar,language-lowcode-1.0.10.jar,language-minecraft-1.0.10.jar -Dfml.gameLayerLibraries=events-1.0.10.jar "-DlegacyClassPath=C:\Users\Joseph\Saved Games\minecraft mods\neoforge\projects\neoforge\.gradle\repositories\ng_dummy_ng\net\minecraft\client\1.20.2\client-1.20.2-client-extra.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.github.oshi\oshi-core\6.4.5\943ba26de047eb6b28fff47f5ee939a34eb5fc8e\oshi-core-6.4.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged.fancymodloader\events\1.0.10\d138400e718426bd9a9a5ec764b3740cbff639b2\events-1.0.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged.fancymodloader\earlydisplay\1.0.10\dc8c52b6092f8407429f6d487bb2be35b76099d0\earlydisplay-1.0.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged.fancymodloader\language-java\1.0.10\a26c7134e83377f8a20f9bfdc78227f64731fe80\language-java-1.0.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged.fancymodloader\language-lowcode\1.0.10\5a5acb3bd79acb4d08cd4d29a08ddfd1680fb35e\language-lowcode-1.0.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged.fancymodloader\language-minecraft\1.0.10\990b15c5a383f9809c57310d061e1dc5c0010fe5\language-minecraft-1.0.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged.fancymodloader\core\1.0.10\12e32e6a1844f17393d0dd70b598217418c41108\core-1.0.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged.fancymodloader\loader\1.0.10\ef686f8601a6d7c22431f49555c3ac7ed12e181\loader-1.0.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged\JarJarSelector\0.4.0\122fa3fe21da1a0ace8323709197616cdb4d1416\JarJarSelector-0.4.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged\JarJarMetadata\0.4.0\6f5e5c52382d11a15f47ca030e046519e389456\JarJarMetadata-0.4.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.google.code.gson\gson\2.10.1\b3add478d4382b78ea20b1671390a858002feb6c\gson-2.10.1.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.google.guava\guava\32.1.2-jre\5e64ec7e056456bef3a4bc4c6fdaef71e8ab6318\guava-32.1.2-jre.jar;C:\Users\Joseph\.m2\repository\com\google\guava\failureaccess\1.0.1\failureaccess-1.0.1.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.ibm.icu\icu4j\73.2\61ad4ef7f9131fcf6d25c34b817f90d6da06c9e9\icu4j-73.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.mojang\authlib\5.0.47\ff7fd2e80a767f6c369fa2065f018ba21d153cbd\authlib-5.0.47.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.mojang\blocklist\1.0.10\5c685c5ffa94c4cd39496c7184c1d122e515ecef\blocklist-1.0.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.mojang\brigadier\1.1.8\5244ce82c3337bba4a196a3ce858bfaecc74404a\brigadier-1.1.8.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.mojang\datafixerupper\6.0.8\3ba4a30557a9b057760af4011f909ba619fc5125\datafixerupper-6.0.8.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.mojang\logging\1.1.1\832b8e6674a9b325a5175a3a6267dfaf34c85139\logging-1.1.1.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.mojang\patchy\2.2.10\da05971b07cbb379d002cf7eaec6a2048211fefc\patchy-2.2.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.mojang\text2speech\1.17.9\3cad216e3a7f0c19b4b394388bc9ffc446f13b14\text2speech-1.17.9.jar;C:\Users\Joseph\.m2\repository\org\apache\httpcomponents\httpclient\4.5.13\httpclient-4.5.13.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\commons-codec\commons-codec\1.16.0\4e3eb3d79888d76b54e28b350915b5dc3919c9de\commons-codec-1.16.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\commons-io\commons-io\2.13.0\8bb2bc9b4df17e2411533a0708a69f983bf5e83b\commons-io-2.13.0.jar;C:\Users\Joseph\.m2\repository\commons-logging\commons-logging\1.2\commons-logging-1.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\io.netty\netty-handler\4.1.97.Final\abb86c6906bf512bf2b797a41cd7d2e8d3cd7c36\netty-handler-4.1.97.Final.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\io.netty\netty-codec\4.1.97.Final\384ba4d75670befbedb45c4d3b497a93639c206d\netty-codec-4.1.97.Final.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\io.netty\netty-transport-classes-epoll\4.1.97.Final\795da37ded759e862457a82d9d92c4d39ce8ecee\netty-transport-classes-epoll-4.1.97.Final.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\io.netty\netty-transport-native-unix-common\4.1.97.Final\d469d84265ab70095b01b40886cabdd433b6e664\netty-transport-native-unix-common-4.1.97.Final.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\io.netty\netty-transport\4.1.97.Final\f37380d23c9bb079bc702910833b2fd532c9abd0\netty-transport-4.1.97.Final.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\io.netty\netty-buffer\4.1.97.Final\f8f3d8644afa5e6e1a40a3a6aeb9d9aa970ecb4f\netty-buffer-4.1.97.Final.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\io.netty\netty-resolver\4.1.97.Final\cec8348108dc76c47cf87c669d514be52c922144\netty-resolver-4.1.97.Final.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\io.netty\netty-common\4.1.97.Final\7cceacaf11df8dc63f23d0fb58e9d4640fc88404\netty-common-4.1.97.Final.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\it.unimi.dsi\fastutil\8.5.12\c24946d46824bd528054bface3231d2ecb7e95e8\fastutil-8.5.12.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.java.dev.jna\jna-platform\5.13.0\88e9a306715e9379f3122415ef4ae759a352640d\jna-platform-5.13.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\ca.weblite\java-objc-bridge\1.1\1227f9e0666314f9de41477e3ec277e542ed7f7b\java-objc-bridge-1.1.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.java.dev.jna\jna\5.13.0\1200e7ebeedbe0d10062093f32925a912020e747\jna-5.13.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged\coremods\6.0.2\74c047103e2268ef9688db18a4df28973f962749\coremods-6.0.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged\neoforgespi\8.0.0\504823c2b4bfec1880b7a80a939af7570ff1862f\neoforgespi-8.0.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged\mergetool\2.0.0\52fe1949be64e3303aabaaa21e315f551db9c9f4\mergetool-2.0.0-api.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged\accesstransformers\9.0.3\3bbbf77a51fcd3fa78751f09a6f40e199a6dd141\accesstransformers-9.0.3.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.neoforged\bus\7.0.17\dfdefadc4603ac9b7a09454107ff80a0c8f1d0df\bus-7.0.17.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\cpw.mods\modlauncher\10.0.9\6d9443f56f50bb85cea383686ff9c867391458b\modlauncher-10.0.9.jar;C:\Users\Joseph\.m2\repository\net\sf\jopt-simple\jopt-simple\5.0.4\jopt-simple-5.0.4.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.apache.commons\commons-compress\1.22\691a8b4e6cf4248c3bc72c8b719337d5cb7359fa\commons-compress-1.22.jar;C:\Users\Joseph\.m2\repository\org\apache\maven\maven-artifact\3.8.5\maven-artifact-3.8.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.apache.commons\commons-lang3\3.13.0\b7263237aa89c1f99b327197c41d0669707a462e\commons-lang3-3.13.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.apache.httpcomponents\httpcore\4.4.16\51cf043c87253c9f58b539c9f7e44c8894223850\httpcore-4.4.16.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.minecraftforge\unsafe\0.2.0\54d7a0a5e8fdb71b973025caa46f341ae5904f39\unsafe-0.2.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.minecrell\terminalconsoleappender\1.2.0\96d02cd3b384ff015a8fef4223bcb4ccf1717c95\terminalconsoleappender-1.2.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.apache.logging.log4j\log4j-core\2.19.0\3b6eeb4de4c49c0fe38a4ee27188ff5fee44d0bb\log4j-core-2.19.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.apache.logging.log4j\log4j-slf4j2-impl\2.19.0\5c04bfdd63ce9dceb2e284b81e96b6a70010ee10\log4j-slf4j2-impl-2.19.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.apache.logging.log4j\log4j-api\2.19.0\ea1b37f38c327596b216542bc636cfdc0b8036fa\log4j-api-2.19.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.joml\joml\1.10.5\22566d58af70ad3d72308bab63b8339906deb649\joml-1.10.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-glfw\3.3.2\757920418805fb90bfebb3d46b1d9e7669fca2eb\lwjgl-glfw-3.3.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-glfw\3.3.2\1251e3cb7e5d6159334cfb9244f789ce992f03b\lwjgl-glfw-3.3.2-natives-windows.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-glfw\3.3.2\e79c4857a887bd79ba78bdf2d422a7d333028a2d\lwjgl-glfw-3.3.2-natives-windows-arm64.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-glfw\3.3.2\17e1f9ec031ef72c2f7825c38eeb3a79c4d8bb17\lwjgl-glfw-3.3.2-natives-windows-x86.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-jemalloc\3.3.2\877e17e39ebcd58a9c956dc3b5b777813de0873a\lwjgl-jemalloc-3.3.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-jemalloc\3.3.2\db886c1f9e313c3fa2a25543b99ccd250d3f9fb5\lwjgl-jemalloc-3.3.2-natives-windows.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-jemalloc\3.3.2\598790de603c286dbc4068b27829eacc37592786\lwjgl-jemalloc-3.3.2-natives-windows-arm64.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-jemalloc\3.3.2\9b07558f81a5d54dfaeb861bab3ccc86bb4477c9\lwjgl-jemalloc-3.3.2-natives-windows-x86.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-openal\3.3.2\ae5357ed6d934546d3533993ea84c0cfb75eed95\lwjgl-openal-3.3.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-openal\3.3.2\e74f299a602192faaf14b917632e4cbbb493c940\lwjgl-openal-3.3.2-natives-windows.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-openal\3.3.2\545ddec7959007a78b6662d616e00dacf00e1c29\lwjgl-openal-3.3.2-natives-windows-arm64.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-openal\3.3.2\21fcb44d32ccf101017ec939fc740197677557d5\lwjgl-openal-3.3.2-natives-windows-x86.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-opengl\3.3.2\ee8e95be0b438602038bc1f02dc5e3d011b1b216\lwjgl-opengl-3.3.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-opengl\3.3.2\83cd34469d4e0bc335bf74c7f62206530a9480bf\lwjgl-opengl-3.3.2-natives-windows.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-opengl\3.3.2\21df035bf03dbf5001f92291b24dc951da513481\lwjgl-opengl-3.3.2-natives-windows-arm64.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-opengl\3.3.2\22fa4149159154b24f6c1bd46a342d4958a9fba1\lwjgl-opengl-3.3.2-natives-windows-x86.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-stb\3.3.2\a2550795014d622b686e9caac50b14baa87d2c70\lwjgl-stb-3.3.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-stb\3.3.2\1c4f4b8353bdb78c5264ab921436f03fc9aa1ba5\lwjgl-stb-3.3.2-natives-windows.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-stb\3.3.2\c29df97c3cca97dc00d34e171936153764c9f78b\lwjgl-stb-3.3.2-natives-windows-arm64.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-stb\3.3.2\a0de7bde6722fa68d25ba6afbd7395508c53c730\lwjgl-stb-3.3.2-natives-windows-x86.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-tinyfd\3.3.2\9f65c248dd77934105274fcf8351abb75b34327c\lwjgl-tinyfd-3.3.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-tinyfd\3.3.2\54a93ed247d20007a6f579355263fdc2c030753a\lwjgl-tinyfd-3.3.2-natives-windows.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-tinyfd\3.3.2\500f5daa3b731ca282d4b90aeafda94c528d3e27\lwjgl-tinyfd-3.3.2-natives-windows-arm64.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl-tinyfd\3.3.2\c1dfa1c438e0262453e7bf625289540e5cbffb2\lwjgl-tinyfd-3.3.2-natives-windows-x86.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl\3.3.2\4421d94af68e35dcaa31737a6fc59136a1e61b94\lwjgl-3.3.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl\3.3.2\a55169ced70ffcd15f2162daf4a9c968578f6cd5\lwjgl-3.3.2-natives-windows.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl\3.3.2\d900e4678449ba97ff46fa64b22e0376bf8cd00e\lwjgl-3.3.2-natives-windows-arm64.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.lwjgl\lwjgl\3.3.2\ed495259b2c8f068794da0ffedfa7ae7c130b3c5\lwjgl-3.3.2-natives-windows-x86.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.slf4j\slf4j-api\2.0.7\41eb7184ea9d556f23e18b5cb99cad1f8581fc00\slf4j-api-2.0.7.jar;C:\Users\Joseph\.m2\repository\com\google\code\findbugs\jsr305\3.0.2\jsr305-3.0.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.jetbrains\annotations\24.0.1\13c5c75c4206580aa4d683bffee658caae6c9f43\annotations-24.0.1.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\cpw.mods\bootstraplauncher\1.1.2\c546e00443d8432cda6baa1c860346980742628\bootstraplauncher-1.1.2.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\cpw.mods\securejarhandler\2.1.10\51e6a22c6c716beb11e244bf5b8be480f51dd6b5\securejarhandler-2.1.10.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.openjdk.nashorn\nashorn-core\15.4\f67f5ffaa5f5130cf6fb9b133da00c7df3b532a5\nashorn-core-15.4.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm-commons\9.5\19ab5b5800a3910d30d3a3e64fdb00fd0cb42de0\asm-commons-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm-util\9.5\64b5a1fc8c1b15ed2efd6a063e976bc8d3dc5ffe\asm-util-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm-analysis\9.5\490bacc77de7cbc0be1a30bb3471072d705be4a4\asm-analysis-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm-tree\9.5\fd33c8b6373abaa675be407082fdfda35021254a\asm-tree-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.ow2.asm\asm\9.5\dc6ea1875f4d64fbc85e1691c95b96a3d8569c90\asm-9.5.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.electronwill.night-config\toml\3.6.4\51d6cefb2b55ee55ee26b16391212fb2c7dfb4f4\toml-3.6.4.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.electronwill.night-config\core\3.6.4\510f174abbf1c947494db50ef2445683bd52c230\core-3.6.4.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.jodah\typetools\0.6.3\a01aaa6ddaea9ec07ec4f209487b7a46a526283a\typetools-0.6.3.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.spongepowered\mixin\0.8.5\9d1c0c3a304ae6697ecd477218fa61b850bf57fc\mixin-0.8.5.jar;C:\Users\Joseph\.m2\repository\com\google\guava\listenablefuture\9999.0-empty-to-avoid-conflict-with-guava\listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.checkerframework\checker-qual\3.33.0\de2b60b62da487644fc11f734e73c8b0b431238f\checker-qual-3.33.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.google.errorprone\error_prone_annotations\2.18.0\89b684257096f548fa39a7df9fdaa409d4d4df91\error_prone_annotations-2.18.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.google.j2objc\j2objc-annotations\2.8\c85270e307e7b822f1086b93689124b89768e273\j2objc-annotations-2.8.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\net.minecraftforge\srgutils\0.4.15\ca408b131759478f164e010fae0d73997e125fb5\srgutils-0.4.15.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.antlr\antlr4-runtime\4.13.1\17125bae1d965624e265ef49552f6465a2bfa307\antlr4-runtime-4.13.1.jar;C:\Users\Joseph\.m2\repository\org\codehaus\plexus\plexus-utils\3.3.0\plexus-utils-3.3.0.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.jline\jline-reader\3.12.1\4382ab1382c7b6f379377ed5f665dc2f6e1218bc\jline-reader-3.12.1.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\com.machinezoo.noexception\noexception\1.7.1\b65330c98e38a1f915fa54a6e5eca496505e3f0a\noexception-1.7.1.jar;C:\Users\Joseph\.gradle\caches\modules-2\files-2.1\org.jline\jline-terminal\3.12.1\c777448314e050d980a6b697c140f3bfe9eb7416\jline-terminal-3.12.1.jar" "-javaagent:C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.1\lib\idea_rt.jar=49318:C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2021.1.1\bin" -Dfile.encoding=UTF-8 @C:\Users\Joseph\AppData\Local\Temp\idea_arg_file1627563801 cpw.mods.bootstraplauncher.BootstrapLauncher --gameDir C:\Users\Joseph\Saved Games\minecraft mods\neoforge\projects\neoforge\run\client --fml.neoForgeVersion 20.2.64-beta --fml.fmlVersion 1.0.10 --fml.mcVersion 1.20.2 --fml.neoFormVersion 20231019.002635 --username Dev --version neoforge --accessToken 0 --userrun mojang --versionrun release --launchTarget forgeclientdev --assetsDir C:\Users\Joseph\.gradle\caches\minecraft\assets\1.20.2 --assetIndex asset-index
[15:51:08.754] [main/INFO] [Launcher/MODLAUNCHER]: ModLauncher running: args [--gameDir, C:\Users\Joseph\Saved, Games\minecraft, mods\neoforge\projects\neoforge\run\client, --fml.neoForgeVersion, 20.2.64-beta, --fml.fmlVersion, 1.0.10, --fml.mcVersion, 1.20.2, --fml.neoFormVersion, 20231019.002635, --username, Dev, --version, neoforge, --accessToken, ❄❄❄❄❄❄❄❄, --userrun, mojang, --versionrun, release, --launchTarget, forgeclientdev, --assetsDir, C:\Users\Joseph\.gradle\caches\minecraft\assets\1.20.2, --assetIndex, asset-index] 
[15:51:08.758] [main/INFO] [Launcher/MODLAUNCHER]: ModLauncher 10.0.9+10.0.9+main.dcd20f30 starting: java version 17.0.8 by Eclipse Adoptium; OS Windows 10 arch amd64 version 10.0 
Exception in thread "main" joptsimple.OptionArgumentConversionException: Cannot parse argument 'C:\Users\Joseph\Saved' of option gameDir
	at MC-BOOTSTRAP/jopt.simple@5.0.4/joptsimple.AbstractOptionSpec.convertWith(AbstractOptionSpec.java:92)
	at MC-BOOTSTRAP/jopt.simple@5.0.4/joptsimple.ArgumentAcceptingOptionSpec.convert(ArgumentAcceptingOptionSpec.java:277)
	at MC-BOOTSTRAP/jopt.simple@5.0.4/joptsimple.OptionSet.valuesOf(OptionSet.java:223)
	at MC-BOOTSTRAP/jopt.simple@5.0.4/joptsimple.OptionSet.valueOf(OptionSet.java:172)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.ArgumentHandler.setArgs(ArgumentHandler.java:50)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.Launcher.run(Launcher.java:86)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.Launcher.main(Launcher.java:78)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26)
	at MC-BOOTSTRAP/cpw.mods.modlauncher@10.0.9/cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23)
	at cpw.mods.bootstraplauncher@1.1.2/cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141)
Caused by: joptsimple.ValueConversionException: Directory [C:\Users\Joseph\Saved] does not exist
	at MC-BOOTSTRAP/jopt.simple@5.0.4/joptsimple.util.PathConverter.convert(PathConverter.java:28)
	at MC-BOOTSTRAP/jopt.simple@5.0.4/joptsimple.util.PathConverter.convert(PathConverter.java:14)
	at MC-BOOTSTRAP/jopt.simple@5.0.4/joptsimple.internal.Reflection.convertWith(Reflection.java:124)
	at MC-BOOTSTRAP/jopt.simple@5.0.4/joptsimple.AbstractOptionSpec.convertWith(AbstractOptionSpec.java:90)
	... 9 more
```

--- a/common/src/main/java/net/neoforged/gradle/common/runs/ide/IdeRunIntegrationManager.java
+++ b/common/src/main/java/net/neoforged/gradle/common/runs/ide/IdeRunIntegrationManager.java
@@ -35,6 +35,7 @@ import java.io.FileWriter;
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.function.Function;
 import java.util.stream.Collectors;
@@ -103,12 +104,9 @@ public class IdeRunIntegrationManager {
                     
                     ideaRun.setMainClass(runImpl.getMainClass().get());
                     ideaRun.setWorkingDirectory(runImpl.getWorkingDirectory().get().getAsFile().getAbsolutePath());
-                    ideaRun.setJvmArgs(String.join(" ", runImpl.realiseJvmArguments()));
+                    ideaRun.setJvmArgs(quoteAndJoin(runImpl.realiseJvmArguments()));
                     ideaRun.moduleRef(project, runIdeaConfig.getPrimarySourceSet().get());
-                    ideaRun.setProgramParameters(runImpl.getProgramArguments().get()
-                                                                          .stream()
-                                                                          .map(arg -> "\"" + arg + "\"")
-                                                                          .collect(Collectors.joining(" ")));
+                    ideaRun.setProgramParameters(quoteAndJoin(runImpl.getProgramArguments().get()));
                     ideaRun.setEnvs(adaptEnvironment(runImpl, RunsUtil::buildRunWithIdeaModClasses));
                     ideaRun.setShortenCommandLine(ShortenCommandLine.ARGS_FILE);
                     
@@ -124,7 +122,11 @@ public class IdeRunIntegrationManager {
             
             
         }
-        
+
+        private static String quoteAndJoin(List<String> args) {
+            return args.stream().map(arg -> "\"" + arg + "\"").collect(Collectors.joining(" "));
+        }
+
         @Override
         public void eclipse(Project project, EclipseModel eclipse) {
             ProjectUtils.afterEvaluate(project, () -> {
@@ -176,7 +178,7 @@ public class IdeRunIntegrationManager {
                 }));
             });
         }
-        
+
         private TaskProvider<?> createIdeBeforeRunTask(Project project, String name, Run run, RunImpl runImpl) {
             final TaskProvider<?> ideBeforeRunTask = project.getTasks().register(CommonRuntimeUtils.buildTaskName("ideBeforeRun", name), task -> {
                 for (SourceSet sourceSet : run.getModSources().get()) {

